internal/testdata/r/terraform_data_resource.tf.tmpl

```diff
@@ -2,7 +2,7 @@
 // changes in resources, not in datasources or outputs.
 // You can pass any [testtemplate.DataCommon] to it.
 
-resource "null_resource" "{{ .RName }}" {
+resource "terraform_data" "{{ .RName }}" {
   triggers = {
     resource = jsonencode({{ .TFID }})
   }
```

Comment:
```suggestion
  input = {{ .TFID }}
```

This should work according to the new docs for testing datasources: https://github.com/hashicorp/terraform-plugin-framework/pull/877/files#diff-577539d2f18c0175da7dcefd0cc00d19c28cfbce5f444b08a3a595c018503d25R108